content/en/docs/components/pipelines/sdk/component-development.md

```diff
@@ -92,299 +106,462 @@ def do_work(input1_file, output1_file, param1):
+      # Path of the program inside the container
+      /pipelines/component/src/program.py,
       --input1-path,
-      {inputPath: Input 1}, # Refers to the "Input 1" input
-
-      --param1,
-      {inputValue: Parameter 1}, # Refers to the "Parameter 1" input
-
-      --output1-path,
-      {outputPath: Output 1}, # Refers to the "Output 1" output
+      {inputPath: Input 1},
+      --param1, 
+      {inputValue: Parameter 1},
+      --output1-path, 
+      {outputPath: Output 1},
     ]
 ```
 
-The above component specification is sufficient, but you should add more
-metadata to make it more useful. The example below includes the following
-additions:
+## Using your component in a pipeline
+
+You can use the Kubeflow Pipelines SDK to load your component using methods
+such as the following:
+
+*   [`kfp.components.load_component_from_file`][kfp-load-comp-file]: 
+    Use this method to load your component from a `component.yaml` path.
+*   [`kfp.components.load_component_from_url`][kfp-load-comp-url]:
+    Use this method to load a `component.yaml` from a URL.
+*   [`kfp.components.load_component_from_text`][kfp-load-comp-text]:
+    Use this method to load your component specification YAML from a string.
+    This method is useful for rapidly iterating on your component
+    specification.
+
+These functions create a factory function that you can use to create
+[`ContainerOp`][kfp-containerop] instances to use as steps in your pipeline.
+This factory function's input arguments include your component's inputs and
+the paths to your component's outputs. The function signature may be modified
+in the following ways to ensure that it is valid and Pythonic.
+
+*   Inputs with default values will come after the inputs without default
+    values and outputs.
+*   Input and output names are converted to Pythonic names (spaces and symbols
+    are replaced with underscores and letters are converted to lowercase). For
+    example, an input named `Input 1` is converted to `input_1`. 
+
+The following example demonstrates how to load the text of your component
+specification and run it in a single-step pipeline. Before you run this
+example, update the component specification to use the component
+specification you defined in the previous sections.
 
-* Component name and description.
-* For each input and output: description, default value, and type.
+```python
+import kfp
+import kfp.components as comp
 
-Final version of `component.yaml`:
+create_step_get_lines = comp.load_component_from_text("""
+name: Get Lines
+description: Gets the specified number of lines from the input file.
 
-```
-name: Do dummy work
-description: Performs some dummy work.
 inputs:
-- {name: Input 1, type: GCSPath, description: 'Data for Input 1'}
-- {name: Parameter 1, type: Integer, default: '100', description: 'Parameter 1 description'} # The default values must be specified as YAML strings.
+- {name: Input 1, type: String, description: 'Data for input 1'}
```

Comment:
The type should not be `String` here as it will interfere with the v2 compiler opinion of what can be done with strings.
Let's use `TextLines` (`Text` would be ideal, but the v2 compiler currently treats is same as `String` - need to fix that first).

Reply:
Actually `Text` and `String` work interchangeably in v2, and I personally think they should.